sdk/search/azure-search/src/main/java/com/azure/search/models/SearchErrorException.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpResponse;
+
+/**
+ * Exception thrown for an invalid response with SearchError information.
+ */
+public final class SearchErrorException extends HttpResponseException {
+    /**
+     * Initializes a new instance of the SearchErrorException class.
+     *
+     * @param message the exception message or the response content if a message is not available.
+     * @param response the HTTP response.
+     */
+    public SearchErrorException(String message, HttpResponse response) {
+        super(message, response);
+    }
+
+    /**
+     * Initializes a new instance of the SearchErrorException class.
+     *
+     * @param message the exception message or the response content if a message is not available.
+     * @param response the HTTP response.
+     * @param value the deserialized response value.
+     */
+    public SearchErrorException(String message, HttpResponse response, SearchError value) {
+        super(message, response, value);
+    }
+
+    @Override
+    public SearchError getValue() {
```

Comment:
I had to add `SearchError` to the Swagger for the sake of JS/TS, but it's my understanding that this is a subset of a well-defined [error contract for Microsoft REST APIs](https://github.com/Microsoft/api-guidelines/blob/master/Guidelines.md#7102-error-condition-responses). Do we actually need to expose this, or is there some Azure Core type for error payloads that overlaps `SearchError`?

Reply:
It's actually useful to have getCode, getMessage etc APIs in SearchError. Otherwise we have to parse the error message.
Autorest will generate exception depending on the name default error (`SearchError` -> `SearchErrorException`) from Swagger.
```
"default": {
            "description": "Error response.",
            "schema": {
              "$ref": "#/definitions/SearchError"
            }
          }
```

In storage, we use the same structure. 
That is expected. What error type do you expect Java use, HttpResponseException?